libwarpaffine/sliceswriter/SlicesWriterTbb.cpp

```diff
@@ -75,6 +79,35 @@ void CziSlicesWriterTbb::WriteWorker()
 
             add_subblock_info.ptrData = sub_block_write_info.add_slice_info.subblock_raw_data->GetPtr();
             add_subblock_info.dataSize = sub_block_write_info.add_slice_info.subblock_raw_data->GetSizeOfData();
+
+            if (this->use_acquisition_tiles_) 
+            {
+                int z;
+                sub_block_write_info.add_slice_info.coordinate.TryGetPosition(libCZI::DimensionIndex::Z, &z);
+                auto guid = this->CreateRetilingIdWithZandSlice(z, sub_block_write_info.add_slice_info.slice_id);
+
+                std::ostringstream oss;
+                oss << "<METADATA><Tags><RetilingId>"
+                    << std::hex << std::uppercase
+                    << std::setw(8) << std::setfill('0') << guid.Data1 << '-'
+                    << std::setw(4) << std::setfill('0') << guid.Data2 << '-'
+                    << std::setw(4) << std::setfill('0') << guid.Data3 << '-'
+                    << std::setw(2) << static_cast<int>(guid.Data4[0])
+                    << std::setw(2) << static_cast<int>(guid.Data4[1]) << '-'
+                    << std::setw(2) << static_cast<int>(guid.Data4[2])
+                    << std::setw(2) << static_cast<int>(guid.Data4[3])
+                    << std::setw(2) << static_cast<int>(guid.Data4[4])
+                    << std::setw(2) << static_cast<int>(guid.Data4[5])
+                    << std::setw(2) << static_cast<int>(guid.Data4[6])
+                    << std::setw(2) << static_cast<int>(guid.Data4[7])
+                    << std::dec
+                    <<"</RetilingId></Tags></METADATA>";
+
+                const string metadata_xml = oss.str();
+                add_subblock_info.ptrSbBlkMetadata = metadata_xml.c_str();
+                add_subblock_info.sbBlkMetadataSize = metadata_xml.size();
```

Comment:
I'd be a bit worried here, because (and this is probably a flaw in libCZI):
* We pass a raw pointer with add_subblock_info.ptrSbBlkMetadata .
* the contract is - the pointer has to be valid during the call to `SyncAddSubBlock `(line 111).

Now - strictly speaking, the pointer we assign is into the variable `metadata_xml`, which goes out-of-scope after the closing curly brace in lime 109. So, I reckon the pointer is to be considered invalid after leaving the scope (and by the time we get into SyncAddSubBlock).

We should ensure that the lifetime of metadata_xml extends beyond the call to SyncAddSubBlock. The simplest way might be:
* move the call to SyncAddSubBlock into the section (i.e. copy line 111 before line 109.
* add an else-section to the if-clause in line 83, and in this else-branch, add another copy of the call to SyncAddSubBlock.

Something like:

```
if (this->use_acquisition_tiles_) 
{
    [....]
    this->writer_->SyncAddSubBlock(add_subblock_info);
}
else
{
    this->writer_->SyncAddSubBlock(add_subblock_info);
}    
```